Marlin/src/gcode/sd/M21_M22.cpp

```diff
@@ -36,9 +37,8 @@ void GcodeSuite::M21() { card.mount(); }
  * M22: Release SD Card
  */
 void GcodeSuite::M22() {
-
   if (!IS_SD_PRINTING()) card.release();
-
+  IF_ENABLED(TFT_COLOR_UI, ui.refresh(LCDVIEW_CALL_REDRAW_NEXT));
```

Comment:
Is here the right place for that?

I prefer that UI code should be handled on UI code, not on core/gcode. 

I think the right place for that, is right after the menu call, isn't?

Reply:
i seen the lcd/marlinui.h in most others M2x commands... and menu code is messy

Reply:
The point is: only when you have that menu opened on screen that you need a refresh, not every time a M22 is called. That is the main reason I would prefer the UI handling its own needs.

Reply:
you only detach the SD from the menu no ? to remove it...

Reply:
you can do it manually calling `M22`... We should try our best to separated core than UI

Reply:
Marlin/src/lcd/menu/menu_main.cpp:159:            GCODES_ITEM(MSG_RELEASE_MEDIA, PSTR("M22"));
Marlin/src/lcd/menu/menu_main.cpp:260:            GCODES_ITEM(MSG_RELEASE_MEDIA, PSTR("M22"));

no way to put code in that...

Reply:
you can change to `ACTION_ITEM` :-)

Reply:
Also, there are other reasons to refresh even on the status page... to show/hide the sd icon

Reply:
Now you got a really good point. So, maybe it's a hook for `card.release`.... I don't know if card.release is called only on M22... This issue happens when we have SD_DETECT and pull out the SD?

Reply:
a refresh doesnt hurt, and you dont do mount/umount during prints

Reply:
I mean: if I pull out the SD card while on that menu screen, it will refresh?

Reply:
for the others, this issue was without SD DETECT PIN, with COLOR UI